code/game/turfs/simulated/minerals.dm

```diff
@@ -732,16 +736,19 @@
 			stage = GIBTONITE_DETONATE
 			explosion(bombturf,1,3,5, adminlog = notify_admins)
 
-/turf/closed/mineral/gibtonite/proc/defuse()
+/turf/closed/mineral/gibtonite/proc/defuse(forceperfect = FALSE)
 	if(stage == GIBTONITE_ACTIVE)
 		cut_overlay(activated_overlay)
 		activated_overlay.icon_state = "rock_Gibtonite_inactive"
 		add_overlay(activated_overlay)
 		desc = "An inactive gibtonite reserve. The ore can be extracted."
 		stage = GIBTONITE_STABLE
-		if(det_time < 0)
+		if(det_time < 0 && !forceperfect)
+			det_time = 0
+			visible_message(span_notice("The chain reaction was stopped! The gibtonite had [det_time] reactions left till the explosion!"))
+		if(forceperfect)
 			det_time = 0
-		visible_message(span_notice("The chain reaction was stopped! The gibtonite had [det_time] reactions left till the explosion!"))
+			visible_message(span_notice("The chain reaction was stopped! The scanner ensured the gibtonite is at its highest potency!"))
 
```

Comment:
You are missing something here for if the det_time is > 0 and it's not forced perfect, using a normal mineral scanner here that you didn't get perfect will result in no message.